File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/AchievementTierResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class AchievementTierResponse 
{
    public long AchievementTierId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public long GoalCount { get; set; }
    public AssetShortResponse Icon { get; set; }
    public long MissionId { get; set; }
    public long GameId { get; set; }
    public long PackId { get; set; }
    public long GameLevelId { get; set; }
    public long GameObjectId { get; set; }
}

File: server-scala-play-server/app/model/AchievementTierResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for AchievementTierResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class AchievementTierResponse(
  achievementTierId: Option[Long],
  title: Option[String],
  description: Option[String],
  goalCount: Option[Long],
  icon: Option[AssetShortResponse],
  missionId: Option[Long],
  gameId: Option[Long],
  packId: Option[Long],
  gameLevelId: Option[Long],
  gameObjectId: Option[Long]
)

object AchievementTierResponse {
  implicit lazy val achievementTierResponseJsonFormat: Format[AchievementTierResponse] = Json.format[AchievementTierResponse]
}

